Correct char.encode_utf16() documentation

tormol opened the pull request:
The "A buffer that's too small" example was calling encode_utf**8**().

alexcrichton commented:
Oh actually I think that this example will fail in this updated case, a 2-element buffer is enough to encode any character in utf-16, so could you change it to a one-element buffer?

steveklabnik commented:
Gah, thank you for this. Agree with @alexcrichton 's suggestion.

tormol commented:
The example still succeeds (at returning `None`) since the buffer is empty.
If I did just that the example would fail since `encode_utf16()` would return `Some(1)`.
The first example already uses a buffer with length 1.

Do you mean changing both examples to use a different char that requires two `u16`s?

alexcrichton commented:
Ah yeah I wasn't sure if `ß` required two u16 elements or not, but yeah the example I believe intends to show one that requires two but the buffer is of length 1

tormol commented:
I have changed both examples to use `𝕊` which requires two `u16`s. (and updated the buffer sizes and expected results accordingly.)

The build failure seems unrelated.

(Do I need to comment after adding commits?)

steveklabnik commented:
> (Do I need to comment after adding commits?)

Yeah, GitHub doesn't notify us when you push :(

(I'll restart Travis just in case)

alexcrichton commented:
@bors: r+ d34c6eeed49acb227bbd07db9d321319cec5dd56

Thanks!
